File: assignment2_q1b.c
#include<stdio.h>
#include<stdlib.h>
int main(){
	

	int no_points = 6*6;
	float pairs = no_points *no_points ;

	float total = 0;

	// Point x co-ordinate
	for (int x1 = 0; x1 < 6; ++x1){
		// Point y co-ordinate
		for (int y1 = 0; y1 < 6; ++y1){


				for (int x2 = 0; x2 < 6; ++x2){
					for (int y2 = 0; y2 < 6; ++y2){

						total = total + abs(x1-x2) + abs(y1-y2);

					}
				}
		}
	}

	float avg = total/pairs;

	printf("The average distance is ");
	printf("%.4f",avg);

}

File: assignment2_q1b.py
points = []

# Assign co-ordinates
for x in range(6):
	for y in range(6):
		points.append([x,y])

no_points = 6*6
pairs = no_points**2

total = 0

for i in range(no_points):
	x1,y1 = points[i]
	for j in range(no_points):
		x2,y2 = points[j]
		total += abs(x2 - x1) + abs(y2 - y1)


avg = total/pairs
print("The average distance between two points of 6x6 grid: ",avg)